test/distributed/org/apache/cassandra/distributed/upgrade/UpgradeTest.java

```diff
@@ -20,13 +20,48 @@
 
 import org.junit.Test;
 
+import org.apache.cassandra.distributed.UpgradeableCluster;
 import org.apache.cassandra.distributed.api.ConsistencyLevel;
 import org.apache.cassandra.distributed.api.Feature;
 import org.apache.cassandra.distributed.shared.Versions;
 import static org.apache.cassandra.distributed.shared.AssertUtils.*;
 
 public class UpgradeTest extends UpgradeTestBase
 {
+    public static void truncateWriteReadTest(UpgradeableCluster cluster, int instanceNum)
```

Comment:
What does the `truncateWriteReadTest` test? Is it for testing the mix-mode messaging filter?

Reply:
Apologies, it was from an earlier iteration where I was trying to prove the issue but it turns out truncate doesn't check the response so it didn't hit the error so I switched to using CAS  for the `Pre40MessageFilterTest`. I'll remove it.